internal/pkg/controllers/selinuxpolicy/configmap_controller.go

```diff
@@ -0,0 +1,290 @@
+		if err := controllerutil.SetControllerReference(cminstance, pod, r.scheme); err != nil {
+			log.Error(err, "Failed to set pod ownership", "pod", pod)
+			return reconcile.Result{}, errors.Wrap(err, "Setting controller reference for SELinux installer pod")
+		}
+
+		// Check if this Pod already exists
+		found := &corev1.Pod{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+		if err != nil && kerrors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+			if err = r.client.Create(context.TODO(), pod); err != nil {
+				return reconcile.Result{}, IgnoreAlreadyExists(err)
+			}
+			continue
+		} else if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "Gettting selinux installer Pod")
+		}
+
+		exitCode, gotCode := r.getInstallerContainerExitCode(found)
+		// Pod is still running - requeue
+		if !gotCode {
+			reqLogger.Info("Pod still running", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+			return reconcile.Result{Requeue: true, RequeueAfter: defaultRequeueTime}, nil
+		}
+		if exitCode != 0 {
+			reqLogger.Info("Pod failed", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name, "exit-code", exitCode)
+			policyCopy.Status.State = spov1alpha1.PolicyStateError
+			if err := r.client.Status().Update(context.TODO(), policyCopy); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "Updating SELinux policy with install error")
+			}
+			return reconcile.Result{}, nil
+		}
+	}
+	policyCopy.Status.State = spov1alpha1.PolicyStateInstalled
+	if err := r.client.Status().Update(context.TODO(), policyCopy); err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "Updating SELinux policy with installation success")
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileConfigMap) getInstallerContainerExitCode(pod *corev1.Pod) (int32, bool) {
+	for idx := range pod.Status.ContainerStatuses {
+		containerStatus := &pod.Status.ContainerStatuses[idx]
+		if containerStatus.Name == "policy-installer" {
+			if containerStatus.State.Terminated != nil {
+				termState := containerStatus.State.Terminated
+				return termState.ExitCode, true
+			}
+		}
+	}
+	return -1, false
+}
+
+// newPodForPolicy returns a busybox pod with the same name/namespace as the cr.
+func newPodForPolicy(name, ns string, node *corev1.Node) *corev1.Pod {
+	labels := map[string]string{
+		"appName":      name,
+		"appNamespace": ns,
+	}
+	trueVal := true
+	hostVolTypeDir := corev1.HostPathDirectory
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetInstallerPodName(name, ns, node),
+			Namespace: GetOperatorNamespace(),
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "policy-installer",
+					// TODO(jaosorior): Add udica image or equivalent to a
+					// more formal registry
+					Image:   "quay.io/jaosorior/udica",
```

Comment:
@JAORMX given how privileged this runs, and that is a direct dependency, it would be great if we could build a version of that image in this repo too, and maybe distribute as `gcr.io/k8s-staging-sp-operator/udica`? 

@saschagrunert @cmurphy any thoughts?

Reply:
Is this the Dockerfile for the image: https://github.com/containers/udica/blob/master/Dockerfile ?

Since the source code is part of a different repository I'm wondering what the best appraoch would be… Maybe we could just integrate the quay.io auto-build into contianers/udica to build the image for each push? I'm concerned about security vulnerabilities, for example in the python packages. If we hit that case, quay will not rebuild if there is no push into the repository, right?

Reply:
@saschagrunert @pjbgf we're actually planning to move away from the udica implementation and have a go-based daemon instead. But yeah, we should figure out a way to deal with external container images.